httpbp/handler.go

```diff
@@ -0,0 +1,74 @@
+package httpbp
+
+import (
+	"context"
+	"html/template"
+	"net/http"
+)
+
+type HandlerFunc func(context.Context, *http.Request, Response) (interface{}, error)
+
+type Middleware func(HandlerFunc) HandlerFunc
+
+func abort(w http.ResponseWriter) {
+	code := http.StatusInternalServerError
+	http.Error(w, http.StatusText(code), code)
+}
+
+func writeResponse(w http.ResponseWriter, resp Response, body interface{}) error {
+	if resp.StatusCode() != 0 && resp.StatusCode() != http.StatusOK {
```

Comment:
nit: `&& resp.StatusCode() != http.StatusOK` part is unnecessary (it will be implicit, but doing that explicitly doesn't hurt anyways).

Reply:
I was thinking that I would just skip the call since `http.StatusOK` is the default and let Go do it's thing.